src/commands/dev/mod.rs

```diff
@@ -46,12 +47,32 @@ pub async fn dev(
         let server_config = server_config.clone();
         async move {
             Ok::<_, failure::Error>(service_fn(move |req| {
-                preview_request(
-                    req,
-                    client.to_owned(),
-                    preview_id.to_owned(),
-                    server_config.clone(),
-                )
+                let client = client.to_owned();
+                let preview_id = preview_id.to_owned();
+                let server_config = server_config.clone();
+                async move {
+                    let resp = preview_request(req, client, preview_id, server_config).await?;
+
+                    let (mut parts, body) = resp.into_parts();
+
+                    let mut headers = HeaderMap::new();
+
+                    for header in &parts.headers {
+                        let (name, value) = header;
+                        let name = name.as_str();
+                        if name.starts_with(HEADER_PREFIX) {
+                            let header_name = &name[HEADER_PREFIX.len()..];
+                            // TODO: remove unwrap
+                            let header_name =
+                                HeaderName::from_bytes(header_name.as_bytes()).unwrap();
```

Comment:
I'd also remove the unwrap in this PR and just wrap whatever the error message is in `failure::Error` (e.g. call `failure::bail!()`)